src/aklite_client_ext.cc

```diff
@@ -0,0 +1,143 @@
+                                              "Apps at boot on a new version of sysroot;"
+                                              " failing current Target: %s, hash: %s") %
+                                current.Name() % current.Sha256Hash())};
+
+      LOG_ERROR << err;
+      client_->notifyTufUpdateFinished(err);
+      // TODO: Specify proper error status
+      return {GetTargetToInstallResult::Status::Failed, TufTarget(), err};
+    }
+    candidate_target = rollback_target;
+    rollback_operation = true;
+    LOG_INFO << "Found Target to rollback to: " << rollback_target.Name() << ", hash: " << rollback_target.Sha256Hash();
+  }
+
+  // This is a workaround for finding and avoiding bad updates after a rollback.
+  // Rollback sets the installed version state to none instead of broken, so there is no
+  // easy way to find just the bad versions without api/storage changes. As a workaround we
+  // just check if the version is not current nor pending nor known (old hash) and never been successfully
+  // installed, if so then skip an update to the such version/Target
+  auto is_bad_target = IsRollback(candidate_target);
+  // Extra state validation
+  if (rollback_operation && is_bad_target) {
+    // We should never get here: when a rollback initiated required, a bad target should never be selected
+    const auto err{
+        boost::str(boost::format("A bad target (%s) was selected for rollback of %s. This should not happen") %
+                   candidate_target.Name() % current.Name())};
+    LOG_ERROR << err;
+    client_->notifyTufUpdateFinished(err);
+    return {GetTargetToInstallResult::Status::Failed, TufTarget(), err};
+  }
+
+  auto res = GetTargetToInstallResult(GetTargetToInstallResult::Status::Ok, candidate_target, "");
+  if (candidate_target.Name() != current.Name() && (!is_bad_target || allow_bad_target)) {
+    if (!rollback_operation && !is_bad_target) {
+      LOG_INFO << "Found new and valid Target to update to: " << candidate_target.Name()
+               << ", sha256: " << candidate_target.Sha256Hash();
+      LOG_INFO << "Updating Active Target: " << current.Name();
+      LOG_INFO << "To New Target: " << candidate_target.Name();
+    } else if (is_bad_target) {
+      // force is true at this point
+      LOG_INFO << candidate_target.Name()
+               << " target is marked for causing a rollback, but installation will be forced ";
+    }
+    // We should install this target:
+    res.selected_target = candidate_target;
+    res.reason = std::string(rollback_operation ? "Rolling back" : "Updating") + " from " + current.Name() + " to " +
+                 res.selected_target.Name();
+
+  } else {
+    if (is_bad_target) {
+      LOG_INFO << "Target: " << candidate_target.Name() << " is a failing Target (aka known locally)."
+               << " Skipping its installation.";
+    }
+
+    auto current_target = std::make_unique<Uptane::Target>(client_->getCurrent());
```

Comment:
Why do we need to get the current at this point if we got one at line 61?

Reply:
Should not be needed. I've just fixed it.